src/_path.h

```diff
@@ -939,79 +939,151 @@ void cleanup_path(PathIterator &path,
     }
 }
 
+void quad2cubic(double x0, double y0,
+                double x1, double y1,
+                double x2, double y2,
+                double *outx, double *outy)
+{
+
+    outx[0] = x0 + 2./3. * (x1 - x0);
+    outy[0] = y0 + 2./3. * (y1 - y0);
+    outx[1] = outx[0] + 1./3. * (x2 - x0);
+    outy[1] = outy[0] + 1./3. * (y2 - y0);
+    outx[2] = x2;
+    outy[2] = y2;
+}
+
+char *__append_to_string(char *p, char *buffer, size_t buffersize,
+                         const char *content)
+{
+    for (const char *i = content; *i; ++i) {
+        if (p < buffer || p - buffer >= (int)buffersize) {
+            return p;
+        }
+
+        *p++ = *i;
+    }
+
+    return p;
+}
+
 template <class PathIterator>
-void convert_to_svg(PathIterator &path,
-                    agg::trans_affine &trans,
-                    agg::rect_d &clip_rect,
-                    bool simplify,
-                    int precision,
-                    char *buffer,
-                    size_t *buffersize)
+void __convert_to_string(PathIterator &path,
+                         int precision,
+                         char **codes,
+                         bool postfix,
+                         char *buffer,
+                         size_t *buffersize)
 {
 #if PY_VERSION_HEX < 0x02070000
     char format[64];
     snprintf(format, 64, "%s.%dg", "%", precision);
 #endif
 
-    typedef agg::conv_transform<py::PathIterator> transformed_path_t;
-    typedef PathNanRemover<transformed_path_t> nan_removal_t;
-    typedef PathClipper<nan_removal_t> clipped_t;
-    typedef PathSimplifier<clipped_t> simplify_t;
-
-    bool do_clip = (clip_rect.x1 < clip_rect.x2 && clip_rect.y1 < clip_rect.y2);
-
-    transformed_path_t tpath(path, trans);
-    nan_removal_t nan_removed(tpath, true, path.has_curves());
-    clipped_t clipped(nan_removed, do_clip, clip_rect);
-    simplify_t simplified(clipped, simplify, path.simplify_threshold());
-
     char *p = buffer;
+    double x[3];
+    double y[3];
+    double last_x = 0.0;
+    double last_y = 0.0;
 
-    const char codes[] = { 'M', 'L', 'Q', 'C' };
-    const int waits[] = { 1, 1, 2, 3 };
-
-    int wait = 0;
+    const int sizes[] = { 1, 1, 2, 3 };
+    int size = 0;
     unsigned code;
-    double x = 0, y = 0;
-    while ((code = simplified.vertex(&x, &y)) != agg::path_cmd_stop) {
-        if (wait == 0) {
-            *p++ = '\n';
 
-            if (code == 0x4f) {
-                *p++ = 'z';
-                *p++ = '\n';
-                continue;
+    while ((code = path.vertex(&x[0], &y[0])) != agg::path_cmd_stop) {
+        if (code == 0x4f) {
+            p = __append_to_string(p, buffer, *buffersize, codes[4]);
+        } else {
+            size = sizes[code - 1];
+
+            for (int i = 1; i < size; ++i) {
+                path.vertex(&x[i], &y[i]);
```

Comment:
It might.  It's certainly possible to create malformed paths from the public API of `path.Path` if you don't know what you're doing.